### PR TITLE
Address miscellaneous issues throughout code base

### PR DESCRIPTION
**Pull request by hiddeco**

### CODE REVIEW

File: internal/decryptor/decryptor.go [hunk 1]
```diff
@@ -508,7 +508,7 @@ func (d *Decryptor) sopsDecryptFile(path string, inputFormat, outputFormat forma
 	if err != nil {
 		return err
 	}
-	err = os.WriteFile(path, out, 0o644)
+	err = os.WriteFile(path, out, 0o600)
```
Comment (stefanprodan) [inline internal/decryptor/decryptor.go:511]: I propose we revert this change so we can merge this PR. We can do this change in dedicated PR and add test to see how it impacts kustomize build.

Comment (hiddeco) [reply internal/decryptor/decryptor.go]: Dropped the commit.

**Review by stefanprodan — APPROVED** [review]
LGTM
